module/mempool/stdmap/backdata/heropool/poolIndex.go

```diff
@@ -0,0 +1,34 @@
+package heropool
+
+// poolIndex represents a slice-based linked list pointer. Instead of pointing
+// to a memory address, this pointer points to a slice index.
+//
+// Note: an "undefined" (i.e., nil) notion for this poolIndex corresponds to the
+// value of uint32(0). Hence, legit "index" values start from uint32(1).
+// poolIndex also furnished with methods to convert a "poolIndex" value to
+// a slice index, and vice versa.
+type poolIndex struct {
+	index uint32
+}
+
+// isUndefined returns true if this poolIndex is set to zero. An undefined
+// poolIndex is equivalent to a nil address-based one.
+func (d poolIndex) isUndefined() bool {
```

Comment:
To match the naming convention of the other functions, why not rename this to `isPoolIndexUndefined` or `isIndexUndefined`?

Reply:
Already replied to / addressed by another comment regarding `poolIndex` vs `EIndex`... :-)